grails-app/controllers/au/org/emii/portal/WpsController.groovy

```diff
@@ -1,31 +1,50 @@
 package au.org.emii.portal
-
-import au.org.emii.portal.proxying.ExternalRequest
 import au.org.emii.portal.proxying.RequestProxyingController
-
 import org.joda.time.DateTime
 
 class WpsController extends RequestProxyingController {
 
     def wpsService
 
+    // Users get WPS status here -> HTML
     def jobReport = {
-        params.url = _getExecutionStatusUrl(params)
-        _performProxyingIfAllowed()
+        params.url = wpsService._getExecutionStatusUrl(params)
+        def execResponse = _performProxyingIfAllowed()
+        if (execResponse != null) {
+            if (execResponse.name() == "ExceptionReport") {
+                _renderExecutionFailed(execResponse)
+            }
+            else if (execResponse.ProcessOutputs.Output.Reference.@href != null){
+                params.status = "Preparing download"
+                _renderExecutionStatus(execResponse)
+            }
+            else {
+                params.status = "Download ready"
+                _renderExecutionStatus(execResponse)
+            }
```

Comment:
Lines 17 - 24 look a bit odd to me; I don't think we need to determine the status here. We can just do it inside `renderExecutionStatus()` which will cut out a bit of code here.